torch_xla/csrc/pooling.cpp

```diff
@@ -348,11 +348,15 @@ xla::XlaOp ComputeMaxPoolIndices(
 
 }  // namespace
 
-bool IsSupportedAdaptiveAvgPool2d(absl::Span<const xla::int64> input_size,
-                                  absl::Span<const xla::int64> output_size) {
+bool IsSupportedAdaptiveAvgPool(absl::Span<const xla::int64> input_size,
+                                absl::Span<const xla::int64> output_size,
+                                int pool_dim) {
   xla::int64 rank = input_size.size();
-  for (int spatial_dim = 0; spatial_dim < 2; ++spatial_dim) {
-    if (input_size[rank - 2 + spatial_dim] % output_size[spatial_dim] != 0) {
+  XLA_CHECK_EQ(output_size.size(), pool_dim)
+      << "Output shape MUST be" << pool_dim;
```

Comment:
Drop this error string. This is more an internal error, actionable by project's developers, more than user.

Reply:
So if someone (user) in their code calls the api with no argument (that's what the test did),  Will they not see this string in the trace?

Reply:
It already does (leveraging the TF/XLA check machinery).

Reply:
Done.